src/build_mozc.py

```diff
@@ -370,18 +371,40 @@ def WriteEnvironmentFile(path, env):
     f.write(nul.join(entries).encode('utf-8'))
 
 
-def UpdateEnvironmentFilesForWindows(out_dir):
+def UpdateEnvironmentFilesForWindows(out_dir, vcvarsall_path):
   """Add required environment variables for Ninja build."""
   python_path_root = MOZC_ROOT
   python_path = os.path.abspath(python_path_root)
   original_python_paths = os.environ.get('PYTHONPATH', '')
   if original_python_paths:
     python_path = os.pathsep.join([original_python_paths, python_path])
+
   for d in os.listdir(out_dir):
     abs_dir = os.path.abspath(os.path.join(out_dir, d))
+    # Tweak generated build rules for ARM64
+    if d.endswith('arm64'):
+      build_ninja = os.path.join(abs_dir, 'build.ninja')
+      with open(build_ninja, 'r', encoding='utf-8') as f:
+        lines = f.readlines()
+        for i in range(0, 2):
+          lines[i] = lines[i].replace('x64\\cl.exe', 'arm64\\cl.exe')
+          lines[i] = lines[i].replace('x86\\cl.exe', 'arm64\\cl.exe')
+      with open(build_ninja, 'w', encoding='utf-8') as f:
+        f.writelines(lines)
+
     for arch in ['x86', 'x64']:
       env_file = os.path.join(abs_dir, f'environment.{arch}')
       env = ReadEnvironmentFile(env_file)
+      # Tweak for ARM64
+      if d.endswith('arm64'):
+        vs_arch = platform.uname().machine
+        if vs_arch != 'arm64':
+          vs_arch = vs_arch + '_arm64'
+        vs_env = get_vs_env_vars(vs_arch, vcvarsall_path_hint=vcvarsall_path)
```

Comment:
This will fail when `stdout` in `get_vsvarsall` is empty.

https://github.com/google/mozc/blob/master/src/build_tools/vs_util.py#L113

Could you add an error check or update `vs_util.py`?

Reply:
Ack. Let's merge https://github.com/google/mozc/pull/1159 first to take care of such a case.

Reply:
Done.